client/src/holdings/AddTradeDialog.tsx

```diff
@@ -38,7 +61,34 @@ export function AddTradeDialog({
   function handleCancel() {
     setOpen(false);
   }
-
+  async function onSubmit(data: CreateTradeData) {
+    const input: CreateTradeData = {
+      holding: holding.id,
+      date: new Date().toISOString().split('T')[0],
+      category: data.category as TradeCategory,
+      broker: data.broker,
+      quantity: data.quantity,
+      priceCurrency: data.priceCurrency,
+      unitPrice: data.unitPrice,
```

Comment:
The values like `data.priceCurrency` have to match the names of the form elements, but they don't atm, e.g. the name is currently 'Unit Price'. So you'll need to change all the names in the JSX to match these ones in camelCase, e.g. change 'Unit Price' to unitPrice